tests/sqlite/sqlite.rs

```diff
@@ -505,6 +507,66 @@ async fn it_can_prepare_then_execute() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[sqlx_macros::test]
+async fn it_can_use_transaction_options() -> anyhow::Result<()> {
+    async fn check_txn_state(
+        conn: &mut SqliteConnection,
+    ) -> Result<SqliteTransactionState, sqlx::Error> {
+        conn.lock_handle().await?.transaction_state()
+    }
+
+    let mut conn = new::<Sqlite>().await?;
+
+    assert_eq!(
+        check_txn_state(&mut conn).await?,
+        SqliteTransactionState::None
+    );
+
+    let mut tx = conn
+        .begin_with(SqliteTransactionBehavior::Deferred.into())
+        .await?;
+    assert_eq!(
+        check_txn_state(&mut *tx).await?,
+        SqliteTransactionState::None
+    );
+    let _ = sqlx::query("SELECT COUNT(*) FROM tweet")
+        .fetch_one(&mut tx)
+        .await?;
+    assert_eq!(
+        check_txn_state(&mut *tx).await?,
+        SqliteTransactionState::Read
+    );
+    let _ = sqlx::query("INSERT INTO tweet ( id, text ) VALUES ( 2, 'Hello, World' )")
+        .execute(&mut tx)
+        .await?;
+    assert_eq!(
+        check_txn_state(&mut *tx).await?,
+        SqliteTransactionState::Write
+    );
+    drop(tx);
+
+    let mut tx = conn
+        .begin_with(SqliteTransactionBehavior::Immediate.into())
+        .await?;
+    assert_eq!(
+        check_txn_state(&mut *tx).await?,
+        SqliteTransactionState::Write
+    );
+    drop(tx);
+
+    // Note: may result in database locked errors if tests are run in parallel
```

Comment:
As suggested here, I'm getting `database is locked` when running the Sqlite test suite around 4 out of 10 times with:
```
DATABASE_URL=sqlite://tests/sqlite/sqlite.db cargo test --no-default-features --features sqlite,all-types,runtime-tokio-native-tls
```
If I run with `-- test-threads 1` I don't get any errors. Just concerned about test flakiness since it doesn't seem that `-- test-threads 1` is the default